test/failure_handler/test/sanity/Suicide.java

```diff
@@ -34,8 +30,7 @@
     public static void main(String[] args) {
         String cmd = null;
         try {
-            String pidStr = ManagementFactory.getRuntimeMXBean().getName()
-                    .split("@")[0];
+            String pidStr = String.valueOf(ProcessHandle.current().pid());
             String osName = System.getProperty("os.name");
             if (osName.contains("Windows")) {
                 cmd = "taskkill.exe /F /PID " + pidStr;
```

Comment:
This can be simplified to ProcessHandle.current().toString().  It returns the pid of the process as a string.

Explicitly converting it to a string is not necessary.  The "+" concatenation would convert the number to a string.

Reply:
Yes, can just have `long pid` in this case.

I don't see that `ProcessHandle.toString` is *specified* to return the string with pid, so relying on that is brittle. We might call `Long.toString` here directly, to avoid jumping through a few calls. But seeing how all this is a test code, that does not seem necessary.

Reply:
@RogerRiggs , as Aleksey pointed out `ProcessHandle::toString` isn't specified to return the string which contains only `pid` (in fact it's not specified at all), so I don't think we should rely on the current implementation.

although it doesn't matter much, I've removed explicit conversion here and replaced `String::valueOf` with `Long::toString` in other places.